lxd/identities.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"context"
+
+	"github.com/canonical/lxd/lxd/db"
+	"github.com/canonical/lxd/lxd/db/cluster"
+	"github.com/canonical/lxd/lxd/identity"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+func updateIdentityCache(d *Daemon) {
+	s := d.State()
+
+	logger.Debug("Refreshing identity cache")
+
+	var identities []cluster.Identity
+	projects := make(map[int][]string)
+	var err error
+	err = s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
```

Comment:
I'm sure there's a context you can use from daemon to cancel the query if the server is shutting down.